util/capacity-reservation.py

```diff
@@ -0,0 +1,104 @@
+#! /usr/bin/env python3
+
+"""Capacity reservation management
+
+Create or cancel AWS EC2 capacity reservations.
+"""
+import argparse
+import boto3
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parent_parser = argparse.ArgumentParser(add_help=False)
+    parent_parser.add_argument(
+        '--aws-profile',
+        help='AWS user profile to use when making boto3 requests')
+    parent_parser.add_argument(
+        '--zone',
+        default='us-east-1a',
+        help='Availability zone to reserve in')
+    parent_parser.add_argument(
+        '--dry-run',
+        help='Show the aws command that would be run, but don\'t run it',
+        action='store_true')
+
+    subparsers = parser.add_subparsers(title='Commands', dest='command')
+    subparsers.required = True
+
+    # Create Command
+    create_parser = subparsers.add_parser(
+        'create',
+        parents=[parent_parser],
+        help='Create a capacity reservation')
+    create_parser.add_argument(
+        '--instance-type',
+        required=True,
+        help='Instance type to reserve')
+    create_parser.add_argument(
+        '--count',
+        required=True,
+        type=int,
+        help='Number of instances to reserve')
+
+    # Cancel Command
+    cancel_parser = subparsers.add_parser(
+        'cancel',
+        parents=[parent_parser],
+        help='Create a capacity reservation')
+    cancel_parser.add_argument(
+        'reservation_id',
+        help='The id of the capacity reservation to be cancelled')
+
+    return parser.parse_args()
+
+
+def create_reservation(client, instance_type, zone, count, dry_run):
+    """Create a reservation for on-demand AWS EC2 instances"""
+    print(
+        'Creating reservation for {} {} in {}'
+        .format(count, instance_type, zone))
+    response = client.create_capacity_reservation(
+        InstancePlatform='Linux/UNIX',
+        InstanceType=instance_type,
+        AvailabilityZone=zone,
+        InstanceCount=count,
+        DryRun=dry_run
```

Comment:
any way we can add tags?

also should we add InstanceMatchCriteria and set it to `targeted`?

Reply:
1. Yes, I can add/require tags. 
2. No, it can't be targeted -- reminder, toil is creating the instances, not us directly, and it doesn't have a way to do that. I would have to edit the toil source / get a PR, redeploy with new toil instance, etc. to make that work

Reply:
Tags would be good, I think we want to track billing for these workflows separately